File: Creational/AbstractFactory/JavaScript/shapeFactory.js
// shapeFactory.js
class ShapeFactory {
  createShape(shapeType) {
    throw new Error("Method 'createShape()' must be implemented.");
  }

  createRenderer() {
    throw new Error("Method 'createRenderer()' must be implemented.");
  }
}

module.exports = ShapeFactory;

File: Creational/AbstractFactory/Python/shape_factory.py
# shape_factory.py

from abc import ABC, abstractmethod
from shape_type import ShapeType
from shape import Shape
from renderer import Renderer

class ShapeFactory(ABC):
  @abstractmethod
  def create_shape(self, shape_type: ShapeType) -> Shape:
    pass

  @abstractmethod
  def create_renderer(self) -> Renderer:
    pass
